src/index.ts

```diff
@@ -10,66 +10,85 @@ import {
   LR_TYPE_PODCAST_EPISODE,
   INITIAL_FACET_STATE
 } from "./constants"
-import { deserializeSearchParams, serializeSearchParams } from "./url_utils"
+import {
+  Facets,
+  deserializeSearchParams,
+  serializeSearchParams
+} from "./url_utils"
 import { useDidMountEffect } from "./hooks"
 
-export const mergeFacetResults = (...args) => ({
+export const emptyOrNil = either(isEmpty, isNil)
+
+type Aggregation = {
+  doc_count_error_upper_bound?: number // eslint-disable-line camelcase
+  sum_other_doc_count?: number // eslint-disable-line camelcase
+  buckets: Array<{ key: string; doc_count: number }> // eslint-disable-line camelcase
+}
+
+type Aggregations = Map<string, Aggregation>
+
+export const mergeFacetResults = (...args: Aggregation[]): Aggregation => ({
   buckets: args
     .map(prop("buckets"))
+    // @ts-ignore
     .reduce((buckets, acc) => unionWith(eqBy(prop("key")), buckets, acc))
 })
 
-export const emptyOrNil = either(isEmpty, isNil)
-
-// take runSearch, clearSearch as callbacks
+// disabling rule here because all functions and values returned by the hook are
+// fully typed, so writing out the explicit return type for this thing is redundant
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export const useCourseSearch = (
-  runSearch,
-  clearSearch,
-  updateURLBar,
-  facets,
-  loaded,
-  searchPageSize
+  runSearch: (
+    text: string,
+    searchFacets: Facets,
+    nextFrom: number
+  ) => Promise<void>,
+  clearSearch: () => void,
+  updateURLBar: (newSearch: string) => void,
+  aggregations: Aggregations,
+  loaded: boolean,
+  searchPageSize: number
 ) => {
   const [incremental, setIncremental] = useState(false)
   const [from, setFrom] = useState(0)
```

Comment:
Does `useState` here need a type?

Reply:
it infers it from the default value - I suppose it could be supplied anyhow, but its smart enough to know that this one should always be a number, the one above a boolean, etc. I supplied it for some of the other ones because I'm using a setter function instead of a literal value.